.github/workflows/dotnet-desktop.yml

```diff
@@ -18,14 +18,35 @@ jobs:
       with:
         fetch-depth: 0
 
-    - name: Setup MSBuild
-      uses: microsoft/setup-msbuild@v1.0.2
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
 
-    - name: Setup VSTest
-      uses: darenm/Setup-VSTest@v1
+    - name: Restore
+      run: dotnet restore -bl:logs/restore.binlog
 
     - name: Build
-      run: msbuild
-      
+      run: dotnet build --configuration Release --no-restore -bl:logs/build.binlog
+
     - name: Test
-      run: vstest.console **\*.UnitTests.dll
+      run: dotnet test --configuration Release --no-build --logger trx --results-directory TestResults
+
+    - name: Upload test results
+      uses: actions/upload-artifact@v4
+      with:
+        name: test-results
+        path: ./TestResults
+      if: ${{ always() }} # Always run this step even on failure
+
+    - name: Upload logs
+      uses: actions/upload-artifact@v4
+      with:
+        name: logs
+        path: ./logs
+      if: ${{ always() }} # Always run this step even on failure
+
+    - name: Upload artifacts
```

Comment:
I don't have any plans to actually ship this tool, I was hoping to keep it so that people needed to clone and build in order to use it.  If that stays that way, we don't need to upload the built artifacts

Reply:
I figured that uploading the artifacts from the pipeline (even if you never do "releases" wouldn't hurt, but I can remove.